rotkehlchen/db/schema.py

```diff
@@ -816,6 +816,23 @@
 );
 """
 
+DB_CREATE_WEB3_NODES = """
+CREATE TABLE IF NOT EXISTS web3_nodes(
+    name TEXT NOT NULL PRIMARY KEY,
+    address TEXT NOT NULL,
+    owned BOOLEAN NOT NULL DEFAULT FALSE,
+    active BOOLEAN NOT NULL DEFAULT FALSE,
+    weight INTEGER NOT NULL
+);
+INSERT OR IGNORE INTO web3_nodes(name, address, owned, active, weight) VALUES ("etherscan", "", FALSE, TRUE, 0.3);
```

Comment:
The reason is that since the users are now able to add new nodes they need to be read from somewhere and that place is this table. If we keep them in a different file there is no easy way to deactivate/delete them or changing their weight. You would need, in order to avoid duplication, a table with the basic information and then some different table to mark them as inactive or deleted.

So the reasons to have all in the database are:
1. Avoid the need of creating a structure that needs to be kept up to date with the nodes (in the ethereum manager)
2. Avoid duplication of information or having it split around files and database
3. Avoid possible errors where info about a node exists but is not in the json file

Reply:
You are already duplicating information. You will need an upgrade each time you want to add a new default node or a data migration.

> Avoid the need of creating a structure that needs to be kept up to date with the nodes (in the ethereum manager)

There is no need for this. You just need to have a data migration for the first time this table is added to the DB (now).

> Avoid duplication of information or having it split around files and database

You already have duplication. You have (1) data in the schema file which is bad on its own. But then you have to duplicate this in the DB upgrade/s.

> Avoid possible errors where info about a node exists but is not in the json file

What errors? This makes no sense. The DB is already the only place you will have data about a node. We are only talking about a place to have the hard-coded default nodes that the DB is initially populated with.

I think the data migration is the place for this.